arangod/Replication2/ReplicatedLog/LogFollower.cpp

```diff
@@ -317,9 +317,10 @@ auto replicated_log::LogFollower::GuardedFollowerData::checkCommitIndex(
     }
   };
 
-  TRI_ASSERT(newLCI >= _largestCommonIndex)
-      << "req.lci = " << newLCI << ", this.lci = " << _largestCommonIndex;
-  if (_largestCommonIndex < newLCI) {
+  // This assertion is no longer true, as followers can now be added.
+  // TRI_ASSERT(newLCI >= _largestCommonIndex)
+  //     << "req.lci = " << newLCI << ", this.lci = " << _largestCommonIndex;
+  if (_largestCommonIndex != newLCI) {
```

Comment:
I see why this can happen. But I think we should prevent this from happening, because the LCI is essentially the index you can use for compaction and you might have removed entries that are then before the LCI.

In particular only the following can happen:
1. The lci is a X.
2. A follower F1 is added, the lci falls to F1.spearhead.index
3. The leader might have already compacted, hence it will override the followers log entirely, starting at X.
4. The LCI will rise again to X.

Reply:
Right. I'll change the condition back to
```suggestion
  // This assertion is no longer true, as followers can now be added.
  // TRI_ASSERT(newLCI >= _largestCommonIndex)
  //     << "req.lci = " << newLCI << ", this.lci = " << _largestCommonIndex;
  if (_largestCommonIndex < newLCI) {
```
to prevent `_largestCommonIndex` from decreasing.